### PR TITLE
Update master branch

### PR DESCRIPTION
**Pull request by DavidSCN**
Open TODOs:
- [x] update wiki and README
- [x] https://github.com/precice/systemtests/pull/228
- [x] https://github.com/precice/tutorials/pull/79
- [x] https://github.com/precice/dealii-adapter/pull/25
- [x] https://github.com/precice/openfoam-adapter/pull/125

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,8 +3,12 @@
     <img src="https://travis-ci.org/precice/dealii-adapter.svg?branch=master" alt="Build status">
 </a>
 
-Coupled structural solvers written with the C++ finite element library deal.II. The coupled codes build on different deal.II tutorials:
-The linear-elastic solver is based on the step-8 tutorial, the nonlinear solver builds on previous work of Jean-Paul Pelteret and Andrew McBride in their deal.II code gallery program 'Quasi-Static Finite-Strain Compressible Elasticity.' Documentation for usage and application can be found in the source code, on the deal.II tutorial documentation and the dealii-adapter wiki pages.
+Coupled structural solvers written with the C++ finite element library deal.II:
+
+- `linear_elasticity` contains a linear-elastic solver based on the step-8 tutorial program of deal.II
+- `nonlinear_elasticity` contains a nonlinear elastic solver, which builds on previous work of Jean-Paul Pelteret and Andrew McBride in their deal.II code gallery program 'Quasi-Static Finite-Strain Compressible Elasticity.' This solver supports shared-memory parallelization.
```
Comment (MakisH) [inline README.md:9]: Citations here would be useful

Comment (DavidSCN) [reply README.md]: I added now links to the respective deal.II tutorial program. Is this, what you had in mind?

**Comment by DavidSCN** [comment]
As reported by @Eder-K , the build fails currently. The reason is, that deal.II added range based for loops, which I used in some several places:

https://github.com/precice/dealii-adapter/blob/5353e6c573c481283f6e3170cb5cd0a7f366084b/linear_elasticity/linear_elasticity.cc#L262-L270

This makes the code (in my opinion) much more readable. In the old version, I wrote:

https://github.com/precice/dealii-adapter/blob/b597eaf5e02b7151f42bfb5a22539a13dcfad3fd/coupled_elasto_dynamics.cc#L613-L623

So, either we switch back to the 'usual' for loop or we indeed need to require version `9.2` instead of `9.1.1`. This is a bit of a conceptual question @uekerman .

**Comment by MakisH** [comment]
Since 9.2.0 has been officially released (20/05/2020), I think we should definitely keep 9.2 and update the system tests.

**Comment by uekerman** [comment]
Just seeing this now. Sry for the late reply. Yes, let's got to v9.2.

File: nonlinear_elasticity/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# To cleanup result files
+SET(CLEAN_UP_FILES
+  # a custom list of globs, e.g. *.log *.vtk
+  *.vtk
+)
+# Usually, you will not need to modify anything beyond this point...
+
+CMAKE_MINIMUM_REQUIRED(VERSION 2.8.12)
+
+FIND_PACKAGE(deal.II 9.1.1 QUIET
```
Comment (uekerman) [inline nonlinear_elasticity/CMakeLists.txt:21]:
```suggestion
FIND_PACKAGE(deal.II 9.2.0 QUIET
```

File: README.md (hunk 1 above)
Comment (DavidSCN) [reply README.md]: @MakisH Can this be resolved?

**Comment by DavidSCN** [comment]
Last open point is now to switch the default build type to release

**Pull request merged**